React/NextJs/ProjectStructure.md

```diff
@@ -30,64 +30,60 @@ src
 .
 └── components
     ├── utilities
-    │   └── meta
+    │   └── Meta
     │       └── Meta.tsx
-    └── ui
-        └── ...ui components...
+    ...
 ```
 ### UI Components
 
-Before implementing [atomic design methodology](https://github.com/danilowoz/react-atomic-design), our common practice was separating components by the scope of the page + core components + shared components like this:
-
-Lowercase folders were indicators of a page scope (except of core).
+When adding UI components, you should be able to distinguish them by:
+1. Page scope based components
+2. Components that are shared all accross the app
+3. Large and complex components which contain a lot of stylings, different layouts and child components.
+   
+Lowercase folders are indicators of a page scoped components, shared components and meta components, while PascalCase folders and filenames should be used for components naming.
 
 ```
+components
 .
-├── src
-└── components
-    ├── core
-    │   └── Button
-    │       ├── index.tsx
-    ├── admin
-    │   └── WelcomeBanner
-    │       └── index.tsx
-    ├── AlbumsCarousel
-    │   └── AlbumsCarousel.tsx
-    └── UserCard
-        └── index.tsx
+├── shared
+│   └── WelcomeBox
+│       └── WelcomeBox.tsx
+├── admin
+│   └── AdminBanner
+│       └── AdminBanner.tsx
+└── Gallery
+    ├── Gallery.tsx
+    └── components
```

Comment:
Hmm, how would the nesting work here? Would we allow nesting of such components or would it only be on an "elements file" basis?

Reply:
I'm not necessary against it, but it should be explicitly defined